pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -68,10 +70,29 @@ def __call__(self, *targs, **tkwargs):
         if callable(obj):
             return self._qfunc_transform(obj, targs, tkwargs)
 
-        raise TransformError(
-            "The object on which the transform is applied is not valid. It can only be a tape, a QNode or a qfunc."
+        # Input is not a QNode nor a quantum tape nor a device.
+        # Assume Python decorator syntax:
+        #
+        # result = some_transform(*transform_args)(qnode)(*qnode_args)
+
+        warnings.warn(
+            "The decorator syntax transform_fn(*transform_args)(qnode) has been "
```

Comment:
just to drive the point home
```suggestion
            "The decorator syntax @transform_fn(*transform_args)(qnode) has been "
```

Reply:
Pretty sure that's not correct syntax. It's either what I've written, or the equivalent
```python
@transform_fn(*transform_args)
def qnode(...):
    ...
```

Reply:
sorry, didn't realize the `(qnode)` at the end. it is indeed not the correct syntax, but then `transform_fn(*transform_args)(qnode)` is not "decorator syntax", so I'd add add the `@` and drop the `(qnode)`

Reply:
I'd like to keep the `qnode` in this part of the warning because one of the alternatives uses it, and it might be confusing if we leave it out in some places and not others. If the issue is because this isn't true decorator syntax, then maybe I can change "The decorator syntax..." into just "The syntax..."?

Reply:
now I'm confused. I think we're talking about 2 different syntaxes, and I also think we're deprecating both of them. I would say "Decorating a QNode with `@transform_fn(*transform_args)` is deprecated"

Reply:
Changed the warning but I had to rephrase some other parts to make it clearer.